ml-agents-envs/mlagents/envs/sampler_class.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np 
+from abc import ABC, abstractmethod
+
+from .exception import SamplerException
+
+class Sampler(ABC): 
+
+    @abstractmethod
+    def sample_parameter(self, *args, **kwargs):
+        pass
+
+
+class UniformSampler(Sampler):
+    def __init__(self, param_dict):
+        self.check_keys(param_dict)
+        self.min_value = param_dict["min"]
+        self.max_value = param_dict["max"]
+
+
+    def check_keys(self, param_dict):
+        keys = ["min", "max"]
+        for key in keys:
+            if key not in param_dict:
+                raise SamplerException(
+                    key + " is required to use Uniform Sampling but" 
+                    " this value isn't provided. Please recheck"
+                    " the documentation to ensure that the sampler file"
+                    " matches the required format."
+                )
+
+    def sample_parameter(self):
+        return np.random.uniform(self.min_value, self.max_value)
```

Comment:
Is there any advantage to using numpy here instead of https://docs.python.org/3/library/random.html#random.uniform ?

Reply:
I decided to use np because I use np.random.choice which allows me to make weighted choices

Reply:
https://docs.python.org/3/library/random.html#random.choices - you can pass the array of weights, or (even better) compute the cumulative weights in `__init__`